openedx/core/djangoapps/credit/api/eligibility.py

```diff
@@ -179,12 +181,17 @@ def get_eligibilities_for_user(username):
     Returns: list
 
     """
+    eligibilities = CreditEligibility.get_user_eligibilities(username)
+
+    if course_key:
+        eligibilities = eligibilities.filter(course_key=course_key)
+
     return [
         {
             "course_key": eligibility.course.course_key,
             "deadline": eligibility.deadline,
         }
-        for eligibility in CreditEligibility.get_user_eligibilities(username)
+        for eligibility in eligibilities
     ]
```

Comment:
@wedaly for now we are using server-to-server API call, we can use parallel server calls but that might affect the loading of page itself. So @tasawernawaz will be implementing cross-domain AJAX calls for getting provider info and eligibility data.